include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -679,7 +678,127 @@ template <typename Lyt, typename TT>
 
     detail::is_operational_impl<Lyt, TT> p{lyt, spec, params};
 
-    return p.determine_operational_input_patterns();
+    std::set<uint64_t> input_patterns{};
+
+    // all possible input patterns
+    for (auto i = 0u; i < spec.front().num_bits(); ++i)
+    {
+        input_patterns.insert(i);
+    }
+
+    const auto non_op_patterns_and_op_status = p.determine_non_operational_input_patterns_and_op_status_due_kinks();
+
+    for (const auto& pattern : non_op_patterns_and_op_status)
```

Comment:
I'd propose using a structured binding here
```suggestion
    for (const auto& [pattern, status] : non_op_patterns_and_op_status)
```